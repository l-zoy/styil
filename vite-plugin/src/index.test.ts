import styilsPlugin from '.'

type Test = {
  transform: (code: string, id: string) => Promise<string>
}
describe('vite plugin', () => {
  it('base', async () => {
    const plugin = styilsPlugin() as Test

    const code = await plugin.transform(
      `
    import { styled } from '@styils/react'

    const Button = styled('button', {})
    `,
      'src/index.ts'
    )

    expect(code).toMatch(/sourceMap/)
  })

  it('base enforce post not .vue', async () => {
    const plugin = styilsPlugin() as Test

    const code = await plugin.transform(
      `
    import { styled } from '@styils/react'

    const Button = styled('button', {})
    `,
      'src/index.vue'
    )

    expect(/sourceMap/.test(code)).toEqual(false)
  })

  it('parameter url', async () => {
    const plugin = styilsPlugin() as Test

    const code = await plugin.transform(
      `
    import { styled } from '@styils/react'

    const Button = styled('button', {})
    `,
      'src/index.ts?test=1'
    )

    expect(code).toMatch(/sourceMap/)
  })

  it('node_modules url', async () => {
    const plugin = styilsPlugin() as Test

    const code = await plugin.transform(
      `
    import { styled } from '@styils/react'

    const Button = styled('button', {})
    `,
      'node_modules/src/index.ts?test=1'
    )

    expect(/sourceMap/.test(code)).toEqual(false)
  })

  it('not babel code', async () => {
    const plugin = styilsPlugin() as Test

    const code = await plugin.transform(
      `
    `,
      'src/index.ts?test=1'
    )

    expect(/sourceMap/.test(code)).toEqual(false)
  })

  it('not babel code', async () => {
    const plugin = styilsPlugin() as Test

    const code = await plugin.transform(
      `
    `,
      'src/index.ts?test=1'
    )

    expect(/sourceMap/.test(code)).toEqual(false)
  })
})
