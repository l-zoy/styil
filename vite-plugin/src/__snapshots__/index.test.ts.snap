// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vite plugin transform .vue file 1`] = `
"
      <script lang=\\"ts\\" setup>
import { ref } from 'vue'
import { styled, useSystem, createGlobal } from './theme'

process.env.NODE_ENV !== 'production' && (styled.sourceMap = \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInNyYy90ZXN0LnZ1ZT9zdHlpbHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBS0EiLCJmaWxlIjoic3JjL3Rlc3QudnVlP3N0eWlscyIsInNvdXJjZXNDb250ZW50IjpbIlxuICAgICAgPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHN0eWxlZCwgdXNlU3lzdGVtLCBjcmVhdGVHbG9iYWwgfSBmcm9tICcuL3RoZW1lJ1xuXG5jb25zdCBCdXR0b24gPSBzdHlsZWQoXG4gICdidXR0b24nLFxuICAobW9kZSkgPT4gKHt9KSxcbilcblxuY3JlYXRlR2xvYmFsKHtcbiAgYm9keToge1xuICAgIGJhY2tncm91bmRDb2xvcjogJ2NhZGV0Ymx1ZSdcbiAgfVxufSlcbjwvc2NyaXB0PlxuICAgICJdfQ== */\\");
const Button = styled(
  'button',
  (mode) => ({}),
)

process.env.NODE_ENV !== 'production' && (createGlobal.sourceMap = \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInNyYy90ZXN0LnZ1ZT9zdHlpbHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBVUEiLCJmaWxlIjoic3JjL3Rlc3QudnVlP3N0eWlscyIsInNvdXJjZXNDb250ZW50IjpbIlxuICAgICAgPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHN0eWxlZCwgdXNlU3lzdGVtLCBjcmVhdGVHbG9iYWwgfSBmcm9tICcuL3RoZW1lJ1xuXG5jb25zdCBCdXR0b24gPSBzdHlsZWQoXG4gICdidXR0b24nLFxuICAobW9kZSkgPT4gKHt9KSxcbilcblxuY3JlYXRlR2xvYmFsKHtcbiAgYm9keToge1xuICAgIGJhY2tncm91bmRDb2xvcjogJ2NhZGV0Ymx1ZSdcbiAgfVxufSlcbjwvc2NyaXB0PlxuICAgICJdfQ== */\\");
createGlobal({
  body: {
    backgroundColor: 'cadetblue'
  }
})
</script>
    "
`;

exports[`vite plugin transform .vue file1 1`] = `
"
      <script lang=\\"ts\\" setup>
import { ref } from 'vue'
import { styled, useSystem, createGlobal } from './theme'

styled(
  'button',
  (mode) => ({}),
)
</script>
    "
`;

exports[`vite plugin transform .vue file2 1`] = `
"
      <script lang=\\"ts\\" setup>
import { ref } from 'vue'
import { styled, useSystem, createGlobal } from './theme'

process.env.NODE_ENV !== 'production' && (styled.sourceMap = \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInNyYy90ZXN0LnZ1ZT9zdHlpbHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBS0EiLCJmaWxlIjoic3JjL3Rlc3QudnVlP3N0eWlscyIsInNvdXJjZXNDb250ZW50IjpbIlxuICAgICAgPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHN0eWxlZCwgdXNlU3lzdGVtLCBjcmVhdGVHbG9iYWwgfSBmcm9tICcuL3RoZW1lJ1xuXG5sZXQgYnV0dG9uID0gc3R5bGVkKFxuICAnYnV0dG9uJyxcbiAgKG1vZGUpID0+ICh7fSksXG4pXG48L3NjcmlwdD5cbiAgICAiXX0= */\\");
let button = styled(
  'button',
  (mode) => ({}),
)
</script>
    "
`;

exports[`vite plugin transform .vue file3 1`] = `
"
      <script lang=\\"ts\\" setup>
import { ref } from 'vue'
import { sx, useSystem, createGlobal } from '@styils/vue'

process.env.NODE_ENV !== 'production' && (sx.sourceMap = \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInNyYy90ZXN0LnZ1ZT9zdHlpbHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBS0EiLCJmaWxlIjoic3JjL3Rlc3QudnVlP3N0eWlscyIsInNvdXJjZXNDb250ZW50IjpbIlxuICAgICAgPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHN4LCB1c2VTeXN0ZW0sIGNyZWF0ZUdsb2JhbCB9IGZyb20gJ0BzdHlpbHMvdnVlJ1xuXG5sZXQgYnV0dG9uID0gc3goXG4gICdidXR0b24nLFxuICAobW9kZSkgPT4gKHt9KSxcbilcbjwvc2NyaXB0PlxuICAgICJdfQ== */\\");
let button = sx(
  'button',
  (mode) => ({}),
)
</script>
    "
`;

exports[`vite plugin transform .vue file4 1`] = `
"
      <script lang=\\"ts\\" setup>
import { ref } from 'vue'
import { sx, useSystem, createGlobal as gl } from '@styils/vue'

process.env.NODE_ENV !== 'production' && (gl.sourceMap = \\"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInNyYy90ZXN0LnZ1ZT9zdHlpbHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBS0EiLCJmaWxlIjoic3JjL3Rlc3QudnVlP3N0eWlscyIsInNvdXJjZXNDb250ZW50IjpbIlxuICAgICAgPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHN4LCB1c2VTeXN0ZW0sIGNyZWF0ZUdsb2JhbCBhcyBnbCB9IGZyb20gJ0BzdHlpbHMvdnVlJ1xuXG5nbCh7fSlcbjwvc2NyaXB0PlxuICAgICJdfQ== */\\");
gl({})
</script>
    "
`;
